libraries/standard/mqtt/src/iot_mqtt_validate.c

```diff
@@ -438,57 +475,103 @@ static bool _validateWildcardHash( uint16_t index,
                      pSubscription->pTopicFilter );
 
         status = false;
-        goto cleanup;
     }
 
-    /* Unless '#' is standalone, it must be preceded by '/'. */
-    if( pSubscription->topicFilterLength > 1 )
+    if( status == true )
     {
-        if( pSubscription->pTopicFilter[ index - 1 ] != '/' )
+        /* Unless '#' is standalone, it must be preceded by '/'. */
+        if( pSubscription->topicFilterLength > 1 )
         {
-            IotLogError( "Invalid topic filter %.*s -- '#' must be preceded by '/'.",
-                         pSubscription->topicFilterLength,
-                         pSubscription->pTopicFilter );
+            if( pSubscription->pTopicFilter[ index - 1 ] != '/' )
+            {
+                IotLogError( "Invalid topic filter %.*s -- '#' must be preceded by '/'.",
+                             pSubscription->topicFilterLength,
+                             pSubscription->pTopicFilter );
 
-            status = false;
-            goto cleanup;
+                status = false;
+            }
         }
     }
 
-cleanup:
-
     return status;
 }
 
 /*-----------------------------------------------------------*/
 
-bool _IotMqtt_ValidateConnect( const IotMqttConnectInfo_t * pConnectInfo )
+static bool _validateClientIdLength( const IotMqttConnectInfo_t * pConnectInfo )
 {
     bool status = true;
     uint16_t maxClientIdLength = MQTT_SERVER_MAX_CLIENTID_LENGTH;
     bool enforceMaxClientIdLength = false;
 
-    /* Check for NULL. */
     if( pConnectInfo == NULL )
```

Comment:
Same here as above. `_IotMqtt_ValidateConnect` already checks pConnectInfo for `NULL`

Reply:
Same comment as above, since it is an independent function now, I added this extra check.